test/Controllers/UsersTest.php

```diff
@@ -69,87 +68,96 @@ public function testCreateUser() {
+        $responseBody = $this->dataFromResponse($response);
+        $this->assertEquals($responseBody, $createUserBody);
     }
 
     /**
      * Test PUT /users/{id}
      * GonebusyLib\Controllers\UsersController::updateUserById()
      */
     public function testUpdateUserById() {
-        // // $this->expectException(EntitiesErrorException::class);
-        //
-        // $id = 'id';
-        // $updateUserByIdBody = new UpdateUserByIdBody();
-        // $response = $this->users->updateUserById(
-        //         $this->authorization,
-        //         $id,
-        //         $updateUserByIdBody);
-        // print_r($response);
-
-        // $this->assertObjectHasAttribute('user', $response);
-        // $this->assertEquals($id, $response->user->accountManagerId);
+        // Create user:
+        $createUserBody = $this->uniqueUserData();
+        $createResponse = $this->users->createUser(Configuration::$authorization, $createUserBody);
+
+        $anotherUserBody = $this->uniqueUserData();
+        # XXX ^ Uses CreateUserBody doubling as UpdateUserByIdBody.
+
+        // Update user by it's id
+        $response = $this->users->updateUserById(
+            Configuration::$authorization,
+            $createResponse->user->id,
+            $anotherUserBody);
+
+        // Was it fetched?
+        $this->assertObjectHasAttribute('user', $response);
+
+        // Does it have all the new data we sent?
+        $responseBody = $this->dataFromResponse($response);
+        $this->assertEquals($responseBody, $anotherUserBody);
     }
 
     /**
      * Test GET /users
      * GonebusyLib\Controllers\UsersController::getUsers()
      */
     public function testGetUsers() {
-
+        $perPage = 3;
         $response = $this->users->getUsers(
-            $this->authorization,
+            Configuration::$authorization,
             $page = 1,
-            $perPage = 3);
-        // print_r($response);
-
-        $this->assertObjectHasAttribute('users', $response);
-
-        // $testingUserBody = new CreateUserBody(
-        //     "8552697701",
-        //     "testing@gonebusy.com",
-        //     "8552697701",
-        //     "6815786092",
-        //     "pro"
-        // );
-        // print_r($testingUserBody);
-        //
-        // $responseBody = new CreateUserBody(
-        //     $response->users[0]->accountManagerId,
-        //     $response->users[0]->email,
-        //     $response->users[0]->id,
-        //     $response->users[0]->resourceId,
-        //     $response->users[0]->role
-        // );
-        // print_r($responseBody);
-        //
-        // $this->assertEquals($testingUserBody, $responseBody);
+            $perPage);
+
+        // Were they fetched?
+        // $this->assertObjectHasAttribute('users', $response);
+        // XXX Is this better? If so apply to all tests:
+        $this->assertInstanceOf('GonebusyLib\Models\GetUsersResponse', $response);
```

Comment:
No, I meant 154: using `assertInstanceOf()` instead of `assertObjectHasAttribute()` (commented line 152). I'm going to go with assertInstanceOf for everything since it's more complete than checking for an assumed object property.

But you're right, my logic agains hardcoding any sort of response structure would also affect the count of `$response->users`... That's harder to fix though so will leave as is per your comment.